tests/fourier/test_qnode_spectrum.py

```diff
@@ -400,20 +400,22 @@ def test_no_generator_expansion(self):
         class _CRX(qml.CRX):
             generator = [None, 1]
 
-        @qml.qnode(dev)
-        def circuit(x, z1, y, z2):
+        @qml.qnode(dev, max_expansion=0)
```

Comment:
So this is to avoid CRX being decomposed now that the stopping condition has changed?

Reply:
Yes, previously, `CRX` was already decomposed in `circuit.construct`, so that the tape expansion would not be triggered based on the missing `generator`. This was key to the bug not being seen by this test case.